nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/util/DataTypeUtils.java

```diff
@@ -2306,4 +2307,47 @@ public static boolean isFittingNumberType(final Object value, final RecordFieldT
 
         return false;
     }
+
+    public static DataType inferSimpleDataType(final String value) {
+        if (value == null || value.isEmpty()) {
+            return null;
+        }
+
+        if (NumberUtils.isParsable(value)) {
```

Comment:
As mentioned in relation to the commons-lang3 dependency declaration, this reference should be replaced with a direct check.  Perhaps implementing a simple check for null and empty, together with moving the Boolean string check prior to this one would be sufficient?  It would also be helpful to add a unit test for this method.